manager/api/jpa/src/main/java/io/apiman/manager/api/jpa/JpaStorage.java

```diff
@@ -2599,12 +2599,15 @@ private void deleteAllContracts(OrganizationBean organizationBean) throws Storag
         String jpql =
             "DELETE FROM ContractBean deleteBean " +
                 "   WHERE deleteBean IN ( " +
-                "       SELECT b " +
-                "           FROM ContractBean b " +
-                "           JOIN b.api apiVersion " +
+                "       SELECT contract " +
```

Comment:
Sure, I try my best to explain:

Consider you have an organization with one client that has contracts to APIs from different organizations:

If all clients are retired it should be save to delete the organization.
During the deletion of the organization the client should be deleted and therefor its contracts (`deleteAllContracts`).
https://github.com/apiman/apiman/blob/bd3542062b0b5ba50e1a755ab6087d10ed8d0dc3/manager/api/jpa/src/main/java/io/apiman/manager/api/jpa/JpaStorage.java#L431-L440


This is currently not possible as the contracts of the client are not cleaned up:
```
Caused by: org.postgresql.util.PSQLException: ERROR: update or delete on table "client_versions" violates foreign key constraint "FK_6h06sgs4dudh1wehmk0us973g" on table "contracts"
  Detail: Key (id)=(3029) is still referenced from table "contracts".
```
The problem is that the existing query only deleted contracts with an API from the org:
https://github.com/apiman/apiman/blob/75eda35aa9ab9608222160081689d1abc1a0ff3f/manager/api/jpa/src/main/java/io/apiman/manager/api/jpa/JpaStorage.java#L2599-L2607

I am not sure if the old behavoir was the desired one, but it avoids easy deletion of an organization until all contracts are manually deleted (even if the client is already retired).

Reply:
Yes, that would explain the existing behavior, at least partially.
On the other side you can delete a single client/api without problems, so these contracts would also disappear.

I agree on the notification.
If I remember correctly a notification is already fired if you retire and API with contracts.

Reply:
Cool, sounds like this is reasonable. 

I will need to give it some extra thought as the blast radius from an error here would be very wide.

Reply:
Yes, no problemo :)
Better to be careful with these changes :skull_and_crossbones:

Reply:
I'm not sure notifications would be fired in this case as we're just operating directly at the data layer and bypass any of  the business logic that would normally trigger those notifications? Is your understanding different?

Reply:
You cannot delete the org with active entities. If you have e.g. a published API you cannot delete it.
So these must be retired manually before org deletion and during that step we send the notification (thats at least what I think it will happen).

I think I would not change so that you can delete the org completely without retire the single entities manually.

I will verify when notifications are really send and come back to this here.

Reply:
Thanks, worth checking whether it's true if the subscriber is in a different org (as outlined previously). I _think_ so, but not 100% certain.

Reply:
Also, it's entirely possible we could provide more "nuclear" options for a future Apiman version.

Reply:
It's a very small addition anyway, so can just be done independently if needed.

I am adding more notifications and events into Apiman 4 anyway.